nexus/src/db/datastore.rs

```diff
@@ -2119,6 +2119,27 @@ impl DataStore {
             })
     }
 
+    pub async fn user_builtin_fetch_by_id(
+        &self,
+        opctx: &OpContext,
+        id: &Uuid,
+    ) -> LookupResult<UserBuiltin> {
+        use db::schema::user_builtin::dsl;
+        opctx.authorize(authz::Action::Read, authz::FLEET.child_generic())?;
```

Comment:
See my top-level note.  I think this will be one of the very few datastore methods that _won't_ have an authz check in it because it will be needed as part of doing authn.  (I think you can keep `pool_authorized(), though.)`